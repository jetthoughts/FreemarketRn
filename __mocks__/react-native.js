const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('../app/api/firebase', () => {
  return {
    saveImageToStorage: jest.fn(),
  }
});

const helpers = require('../app/lib/helpers');
helpers.dateToTimeString = () => '10:20';

module.exports = rn;
