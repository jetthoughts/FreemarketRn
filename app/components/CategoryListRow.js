import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, TouchableHighlight, Text, View } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

const textColor = "white";
const styles = StyleSheet.create({
  row: {
    flex: 1,
    flexDirection: 'row',
  },
  text: {
    height: 40,
    fontSize: 18,
    paddingLeft: 10
  }
});

const CategoryListRow = ({ onPress, name }) => (
  <TouchableHighlight onPress={onPress} >
    <View style={styles.row}>
      <Icon name="book" size={24} color="darkgray" />
      <Text style={styles.text}>{name}</Text>
    </View>
  </TouchableHighlight>
);

CategoryListRow.propTypes = {
  name: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

export default CategoryListRow;
