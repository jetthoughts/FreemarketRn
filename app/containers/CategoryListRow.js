import React, { Component } from 'react';
import { TouchableHighlight, Text } from 'react-native';

export default class CategoryListRow extends Component {
  onPressCategory() {
    return this.props.onPressCategory(this.props.rowID);
  }

  render() {
    return (
      <TouchableHighlight onPress={() => this.onPressCategory()} >
        <Text>{this.props.rowData.name}</Text>
      </TouchableHighlight>
    );
  }
}
