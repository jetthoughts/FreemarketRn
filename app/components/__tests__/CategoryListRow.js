import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import CategoryListRow from '../CategoryListRow';

it('renders correctly', () => {
  const tree = renderer.create(
    <CategoryListRow
      onPress={() => true}
      name="Item"
      icon="book"
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
