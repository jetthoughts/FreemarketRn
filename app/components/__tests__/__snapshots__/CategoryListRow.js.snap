// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "darkgray",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï€­
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "height": 40,
          "paddingLeft": 10,
        }
      }
    >
      Item
    </Text>
  </View>
</View>
`;
